services/src/main/java/org/keycloak/services/resources/admin/info/ServerInfoAdminResource.java

```diff
@@ -196,6 +191,23 @@ private void setThemes(ServerInfoRepresentation info) {
             }
         }
     }
+    
+    private LinkedList<String> filterThemes(Theme.Type type, LinkedList<String> themeNames) {
+        LinkedList<String> filteredNames = new LinkedList<>(themeNames);
+        
+        boolean filterAccountV2 = (type == Theme.Type.ACCOUNT) && 
+                !Profile.isFeatureEnabled(Profile.Feature.ACCOUNT2);
+        boolean filterAdminV2 = (type == Theme.Type.ADMIN) && 
+                !Profile.isFeatureEnabled(Profile.Feature.ADMIN2);
+        
+        if (filterAccountV2 || filterAdminV2) {
+            filteredNames.remove("keycloak.v2");
+            filteredNames.remove("rh-sso.v2");
+            return filteredNames;
```

Comment:
This return statement seems unnecessary as this value is also returned on the last line of the method.

Reply:
Fixed